crates/cdk-ldk-node/src/web/handlers/channels.rs

```diff
@@ -18,6 +18,73 @@ use crate::web::templates::{
     error_message, form_card, format_sats_as_btc, info_card, layout, success_message,
 };
 
+// Node alias cache for consistent display
+use std::sync::Arc;
+use tokio::sync::RwLock;
+use std::collections::HashMap as StdHashMap;
+
+lazy_static::lazy_static! {
+    static ref NODE_ALIAS_CACHE: Arc<RwLock<StdHashMap<String, String>>> = Arc::new(RwLock::new(StdHashMap::new()));
+}
+
+/// Fetch node alias from external sources
+async fn get_node_alias(node_id: &str) -> Option<String> {
+    // Check cache first
+    {
+        let cache = NODE_ALIAS_CACHE.read().await;
+        if let Some(alias) = cache.get(node_id) {
+            return Some(alias.clone());
+        }
+    }
+
+    // Try to fetch from 1ml.com API (for mainnet/testnet)
+    let alias = fetch_node_alias_from_1ml(node_id).await;
+
+    if let Some(alias) = &alias {
+        // Cache the result
+        let mut cache = NODE_ALIAS_CACHE.write().await;
+        cache.insert(node_id.to_string(), alias.clone());
+        return Some(alias.clone());
+    }
+
+    // Fallback: Generate a test alias for unknown nodes (useful for testnet/regtest)
+    let test_alias = generate_test_alias(node_id);
+    let mut cache = NODE_ALIAS_CACHE.write().await;
+    cache.insert(node_id.to_string(), test_alias.clone());
+    Some(test_alias)
+}
+
+/// Fetch node alias from 1ml.com API
+async fn fetch_node_alias_from_1ml(node_id: &str) -> Option<String> {
+    let client = reqwest::Client::new();
+    let url = format!("https://1ml.com/node/{}/json", node_id);
+
+    match client.get(&url).send().await {
+        Ok(response) => {
+            if response.status().is_success() {
+                match response.json::<serde_json::Value>().await {
+                    Ok(json) => {
+                        if let Some(alias) = json.get("alias").and_then(|v| v.as_str()) {
+                            return Some(alias.to_string());
+                        }
+                    }
+                    Err(_) => {}
+                }
+            }
+        }
+        Err(_) => {}
+    }
+
+    None
+}
```

Comment:
Can we take this out? I know I asked for it and its a nice feature but I thought we could get the alias locally I would rather not introduce this call to a third party at least for now maybe we can revisit.